lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,49 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    query_string = ""
+    ctx_main_script = ""
+    if ctx:
+        query_string = ctx.query_string
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+    main_script_directory = os.path.dirname(main_script_path)
+
+    if page.startswith("./"):
+        page = page[2:]
+    elif page.startswith("/"):
+        page = page[1:]
+
+    requested_page = os.path.join(main_script_directory, page)
+    all_app_pages = source_util.get_pages(ctx_main_script).values()
+
+    for page_data in all_app_pages:
+        full_path = page_data["script_path"]
+
+        if requested_page == full_path:
+            raise RerunException(
+                RerunData(
+                    query_string=query_string,
```

Comment:
Perhaps we can talk with @willhuang1997 and @sfc-gh-jcarroll but do we not want the query string to pass along now? Let's just make sure these work together.

Reply:
Just looking at the code, I would guess that the query string will be `""` because the `page_script_hash` should differ. As a result, I think this code is fine. I think an automated test with query string would be appreciated tho within the e2e test to check that the url for query_params are not retained.

Reply:
Went to write the test and even with the change `RerunData(query_string=""`  the query params are still preserved between pages - any thoughts?

Reply:
hmmm it's because https://github.com/streamlit/streamlit/pull/7774 is not merged yet. I think we should maybe merge that one first and then merge develop into this branch? If you want to test it locally, you can merge the branch above into your local branch.